### PR TITLE
add error handling and filter to get config requests

### PR DESCRIPTION
**Pull request by rumenvasilev**
This adds a very basic error propagation support, so errors could at least be seen. It's pretty basic, but I guess for the alpha version it's better than nothing. Here's example output:
```
2023/02/21 23:34:40 failed to get config: rpc call returned errors
type: protocol, tag: operation-failed, appTag: , path: , message: syntax error
type: protocol, tag: operation-failed, appTag: , path: , message: syntax error
```

The second change is adding filter support to `GetConfig()`. The usage becomes something like this:

`GetConfig()`, without filter:
```xml
<configuration xmlns="http://xml.juniper.net/xnm/1.1/xnm" junos:commit-seconds="1676994552" junos:commit-localtime="2023-02-21 15:49:12 UTC" junos:commit-user="root">
    <version>20200819.234446.1_builder.r1131461</version>
    <system>
        <host-name>csrx-netconf-dut</host-name>
        <root-authentication>
        <junos:comment># juniper123</junos:comment>
            <encrypted-password>$6$dulWt$kKfwWgbSMtvn38k6fJYRVqIaMe6FMfYTMBmIE.oI8a0Z4Xe9Lj5zK/TVWlT140JCNsNbm4A2KHj3UkRWcDtV41</encrypted-password>
        </root-authentication>
        <services>
            <ssh>
                <root-login>allow</root-login>
                <ciphers>aes128-ctr</ciphers>
                <ciphers>aes256-ctr</ciphers>
            </ssh>
            <netconf>
                <junos:comment># This uses normal ssh port 22 and not port 830</junos:comment>
                <ssh>
                </ssh>
            </netconf>
        </services>
    </system>
    <interfaces>
        <interface>
            <name>ge-0/0/0</name>
            <unit>
                <name>0</name>
                <family>
                    <inet>
                        <address>
                            <name>172.19.0.2/24</name>
                        </address>
                    </inet>
                </family>
            </unit>
        </interface>
    </interfaces>
    <security>
        <zones>
            <security-zone>
                <name>test1</name>
                <tcp-rst/>
            </security-zone>
            <security-zone>
                <name>test2</name>
                <tcp-rst/>
            </security-zone>
        </zones>
    </security>
</configuration>
```

`GetConfig()` with filter `security/zones/security-zone/name=test1`:
```xml
<configuration xmlns="http://xml.juniper.net/xnm/1.1/xnm" junos:commit-seconds="1676994552" junos:commit-localtime="2023-02-21 15:49:12 UTC" junos:commit-user="root">
    <security>
        <zones>
            <security-zone>
                <name>test1</name>
                <tcp-rst/>
            </security-zone>
        </zones>
    </security>
</configuration>
```

`GetConfig()` with filter `security/zones/security-zone` (without selector):
```xml
<configuration xmlns="http://xml.juniper.net/xnm/1.1/xnm" junos:commit-seconds="1676994552" junos:commit-localtime="2023-02-21 15:49:12 UTC" junos:commit-user="root">
    <security>
        <zones>
            <security-zone>
                <name>petko-bika</name>
                <tcp-rst/>
                <enable-reverse-reroute/>
            </security-zone>
            <security-zone>
                <name>test1</name>
                <tcp-rst/>
            </security-zone>
            <security-zone>
                <name>test2</name>
                <tcp-rst/>
            </security-zone>
        </zones>
    </security>
</configuration>
```

Trailing `/` are ignored in the filter as are empty paths, for example `this/filter/here` is the same as `this//filter/here/`.

### CODE REVIEW

**Comment by rumenvasilev** [comment]
@nemith feel free to yell at me for whatever you don't like. This is a draft PR for that same reason :)
I'd like to know if structurally it is OK with you and will proceed with adding tests afterwards.

I was actually relying on all the tests automatically kicking in, but meh - needs your approval even for that.

File: session.go [hunk 1]
```diff
@@ -301,6 +302,14 @@ func (s *Session) Call(ctx context.Context, op interface{}, resp interface{}) er
 		return err
 	}
 
+	if len(reply.Errors) > 0 {
+		err := errors.New("rpc call returned errors")
+		for _, v := range reply.Errors {
+			err = errors.Join(err, fmt.Errorf("type: %s, tag: %s, appTag: %s, path: %s, message: %s", v.Type, v.Tag, v.AppTag, v.Path, v.Message))
```
Comment (nemith) [inline session.go:308]: errors.Join only came out in 1.20 and we need to at least support 1.19 and maybe 1.18.  

But i do think that we should use it, but maybe backport it for older versions and guard it with build tags (it's pretty simple)

Also this should not be a fmt.Errorf which is just a string but use this type: https://github.com/nemith/netconf/blob/main/msg.go#L59

The Error method on this type makes it a Go error type that can be used.

File: session.go [hunk 2]
```diff
@@ -301,6 +302,14 @@ func (s *Session) Call(ctx context.Context, op interface{}, resp interface{}) er
 		return err
 	}
 
+	if len(reply.Errors) > 0 {
+		err := errors.New("rpc call returned errors")
```
Comment (nemith) [inline session.go:306]: I am not sure this error helps at all.  The specific RPC errors should be good enough

File: session.go [hunk 3]
```diff
@@ -301,6 +302,14 @@ func (s *Session) Call(ctx context.Context, op interface{}, resp interface{}) er
 		return err
 	}
 
+	if len(reply.Errors) > 0 {
```
Comment (nemith) [inline session.go:305]: This should be a separate PR.
